dvc/dependency/repo.py

```diff
@@ -78,9 +78,21 @@ def download(self, to, jobs=None):
             if self.def_repo.get(self.PARAM_REV_LOCK) is None:
                 self.def_repo[self.PARAM_REV_LOCK] = repo.get_rev()
 
-            _, _, cache_infos = repo.fetch_external([self.def_path], jobs=jobs)
-
-        cache.checkout(to.path_info, cache_infos[0])
+            path_info = PathInfo(repo.root_dir) / self.def_path
+            hash_info = repo.repo_tree.get_hash(
+                path_info, follow_subrepos=False, jobs=jobs,
+            )
+            if not hash_info:
+                raise FileNotFoundError
+            cache.save(
+                path_info,
+                repo.repo_tree,
+                hash_info,
+                link=False,
+                save_link=False,
+            )
```

Comment:
This is really straight-to-cache functionality that we are currently working on. Put it here in an explicit form for simplicity, will likely generalise and move this in this PR.